### ISSUE TITLE
TypeScript definition for WebsocketClient doesn't match the actual arguments

### ISSUE THREAD

**Issue by ncallaway** [coinbase/gdax-node#169]
The exported TypeScript definition for the constructor of WebsocketClient only declares a single argument of `productIds`. The actual constructor takes several arguments, including `productIDs`, `websocketURI`, `auth`, and an object describing the channels to subscribe to.

**TypeScript Definition**
https://github.com/coinbase/gdax-node/blob/9a59db269857d10cb449c44cd6a0f51c630abd5c/index.d.ts#L243

```
    export class WebsocketClient {
        constructor(productIds: string[]);

        on(event: 'message', eventHandler: (data) => void);
        on(event: 'error', eventHandler: (err) => void);
        on(event: 'open', eventHandler: () => void);
        on(event: 'close', eventHandler: () => void);
    }
```

**Websocket Constructor**

https://github.com/coinbase/gdax-node/blob/9a59db269857d10cb449c44cd6a0f51c630abd5c/lib/clients/websocket.js#L13

 ```
 constructor(
    productIDs,
    websocketURI = 'wss://ws-feed.gdax.com',
    auth = null,
    { heartbeat = false, channels = null } = {}
  )
```

**Expected Behavior**
When using TypeScript, we should be able to construct a WebSocket client that can pass in the necessary parameters, including `websocketURI`, `auth`, `heartbeat` and `channels`.

Example:
```
const establishedSocket = new Gdax.WebsocketClient(
      ["BTC-USD"],
      api.wsEndpoint,
      { key: sandbox.key, secret: sandbox.secret, passphrase: sandbox.passphrase },
      { channels: ["ticker", "user"] }
    );
```

**Actual Behavior**
When using TypeScript, if any parameters aside from `productIDs` are provided, a TypeScript compile error occurs.

Error:
`TS2554: Expected 1 arguments, but got 4.`

**Proposed Fix**
Update the TypeScript definition to match the actual constructor of the WebsocketClient. I'll have a pull request available with this proposed fix shortly.

**Comment by ncallaway** [comment]
This issue would also be addressed by pull request #160.

**Comment by fb55** [comment]
Fixed in #160
